src/utils/transactionHelper.ts

```diff
@@ -0,0 +1,57 @@
+import { fromNullable } from 'fp-ts/lib/Option';
+import { fromLeft, fromPredicate, taskEither, TaskEither, tryCatch } from 'fp-ts/lib/TaskEither';
+import { Client } from '../../generated/definitions/pagopa/client';
+import { Transaction } from '../../generated/definitions/pagopa/Transaction';
+import { TransactionStatusResponse } from '../../generated/definitions/pagopa/TransactionStatusResponse';
+import { GENERIC_STATUS, TX_ACCEPTED, UNKNOWN } from './TransactionStatesTypes';
+
+export const checkStatusTask = (
+  transactionId: string,
+  paymentManagerClient: Client,
+): TaskEither<UNKNOWN, TransactionStatusResponse> =>
+  tryCatch(
+    () =>
+      paymentManagerClient.checkStatusUsingGET({
+        id: transactionId,
+      }),
+    () => UNKNOWN.value,
+  ).foldTaskEither(
+    err => fromLeft(err),
+    errorOrResponse =>
+      errorOrResponse.fold(
+        () => fromLeft(UNKNOWN.value),
+        responseType => (responseType.status !== 200 ? fromLeft(UNKNOWN.value) : taskEither.of(responseType.value)),
+      ),
+  );
+
+export const getDataFromSessionStorageTask = (key: string): TaskEither<UNKNOWN, Transaction> =>
+  Transaction.decode(JSON.parse(fromNullable(sessionStorage.getItem(key)).getOrElse(''))).fold(
+    _ => fromLeft(UNKNOWN.value),
+    transaction => taskEither.of(transaction),
+  );
+
+export const isNot3dsFlowTask = (
+  transactionStatusResponse: TransactionStatusResponse,
+): TaskEither<UNKNOWN, TransactionStatusResponse> =>
+  fromPredicate(
+    (transaction: TransactionStatusResponse) => fromNullable(transaction.data?.acsUrl).isNone(),
+    _ => UNKNOWN.value,
+  )(transactionStatusResponse);
+
+export const showErrorStatus = () => {
```

Comment:
@infantesimone Maybe we should move rendering helpers to a different module

Reply:
I agree, for now I can move it to response.pug?